src/test/java/LoginTests.java

```diff
@@ -1,28 +1,43 @@
 import org.openqa.selenium.By;
 import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.PageFactory;
 import org.testng.Assert;
 import org.testng.annotations.Test;
+import pages.LoginPage;
 
 public class LoginTests extends BaseTest {
+    LoginPage loginPage = new LoginPage();
+
+    @Test(dataProvider = "IncorrectLoginProviders", dataProviderClass = BaseTest.class)
+    public void negativeLoginTests(String email, String password) throws InterruptedException {
+        PageFactory.initElements(basePage.getDriver(), loginPage);
+        loginPage.login(email, password);
+        Thread.sleep(5000);
```

Comment:
Please, delete `throws InterruptedException`

**Thread. sleep is bad! It blocks the current thread and renders it unusable for further work.
Using Thread. sleep() frequently in an automation framework is not a good practice.**

Please use the Explicit Wait command, the WebDriver is directed to wait until a certain condition occurs before proceeding with executing the code.